src/gui/MidiClipView.cpp

```diff
@@ -140,8 +142,64 @@ void MidiClipView::changeName()
 
 
 
+void MidiClipView::transposeSelection()
+{
+	const auto selection = getClickedClips();
+
+	// Calculate the key boundries for all clips
+	int highest = 0;
+	int lowest = NumKeys - 1;
+	for (ClipView* clipview: selection)
+	{
+		if (auto mcv = dynamic_cast<MidiClipView*>(clipview))
+		{
+			TimePos start, end;
+			int low, high;
+			if (mcv->getMidiClip()->notes().getBounds(start, end, low, high))
+			{
+				lowest = std::min(low, lowest);
+				highest = std::max(high, highest);
+			}
+		}
+	}
+
+	int minStep = 0 - lowest;
+	int maxStep = NumKeys - 1 - highest;
+	int semitones = QInputDialog::getInt(this, tr("Transpose"), tr("Semitones to transpose:"), 0, minStep, maxStep, 1);
```

Comment:
Personally, I would have written "Semitones to transpose *by*:", but I'll leave it up to you.

Reply:
You're the better English speaker!